rules/alerting.go

```diff
@@ -311,7 +311,9 @@ func (r *AlertingRule) Eval(ctx context.Context, ts time.Time, query QueryFunc,
 	resultFPs := map[uint64]struct{}{}
 
 	var vec promql.Vector
-	for _, smpl := range res {
+	for i := range res {
```

Comment:
Why this change?

Reply:
To get the pointer and be able to change the sample in place

Reply:
What's that got to do with this PR?

Reply:
I need to alter the 'res' vector to look for common labels

Reply:
I'm not seeing it, ContainsSameLabelset doesn't alter the vector.

Reply:
It does not but we must have a vector to run it against.

Reply:
Ah, I see what you're up to now.